SS2-Project/Assets/Starstorm2/ScriptableObjects/SS2AssetCollections.cs

```diff
@@ -8,19 +8,70 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEditor;
+using System.Reflection;
+using HG;
+using UnityEngine.Networking;
+using RoR2.Projectile;
+using System.IO;
+using Path = System.IO.Path;
+using System.Linq;
 namespace SS2
 {
-    //todo: a lot
+    public class ExtendedAssetCollection : AssetCollection
+    {
+        public void OnValidate()
+        {
+            //very slow lol
+
+            //ensure field values are contained in AssetCollection.assets
+            foreach (FieldInfo f in this.GetType().GetFields())
```

Comment:
From what i understand, this ensures that the specific extended asset collection fields are in the .assets array.

This is... yeah quite slow, i think it might be better to have something like this maybe

```cs
public class ExtendedAssetCollection : AssetCollection
{
   internal virtual string[] extraFieldNames {get;
}

public class SomeOtherAssetCollection : ExtendedAssetCollection
{
   override string[] extraFieldNames = new string[] { nameof(someSpecificAsset) }
   public GameObject someSpecificAsset;
}```

This way we can just use the Type.GetField() method instead of obtaining all the fields and then iterating thru them.